elements/rh-avatar/rh-avatar.ts

```diff
@@ -1,32 +1,303 @@
 
-import { pfelement } from '@patternfly/pfe-core/decorators.js';
+import { BaseAvatar } from '@patternfly/pfe-avatar/BaseAvatar.js';
+
+import { hash } from './lib/djb-hash';
+import { hsl2rgb, rgb2hsl } from './lib/hslrgb';
 
 import styles from './rh-avatar.css';
 
+import { colorContextProvider } from '../../lib/context/color.js';
+
+type Vector2D = [x: number, y: number];
+type Colors = Record<`color${number}`, string>;
+
+/** Ensure avatar colours are registered before custom elements upgrade */
+function register(klass: typeof RhAvatar) {
+  klass.registerColors();
+}
+
 /**
  * An avatar is a visual used to represent a user. It may contain an image or
  * a placeholder graphic.
  *
  *
  * @summary  An avatar is a visual used to represent a user.
  */
-@customElement('rh-avatar') @pfelement()
-export class RhAvatar extends LitElement {
+@customElement('rh-avatar')
+export class RhAvatar extends BaseAvatar {
   static readonly version = '{{version}}';
 
   static readonly styles = [styles];
 
+  private static readonly defaultSize = 128;
+
+  private static readonly defaultColors = '#67accf #448087 #709c6b #a35252 #826cbb';
+
+  private static _registerColor(color: RGBTriple): void {
+    RhAvatar.colors.push({
+      color1: `rgb(${color.join(',')})`,
+      color2: `rgb(${this._adjustColor(color).join(',')})`,
+    });
+  }
+
+  private static _adjustColor(color: RGBTriple): RGBTriple {
+    const dark = 0.1;
+    const lAdj = 0.1; // luminance adjustment
+    const hsl = rgb2hsl(...color);
+
+    // if luminance is too dark already, then lighten the alternate color
+    // instead of darkening it.
+    hsl[2] += hsl[2] > dark ? -lAdj : lAdj;
+
+    return hsl2rgb(...hsl);
+  }
+
+  static colors: Colors[] = [];
+
+  static customColors: string;
+
+  static registerColors() {
+    RhAvatar.colors = [];
+
+    const contextColors = this.customColors || this.defaultColors;
+
+    contextColors.split(/\s+/).forEach(colorCode => {
+      let pattern;
+      switch (colorCode.length) {
+        case 4: // ex: "#0fc"
+          pattern = /^#([A-f0-9])([A-f0-9])([A-f0-9])$/.exec(colorCode);
+          if (pattern) {
+            pattern.shift();
+            const color = pattern.map(c => parseInt(c + c, 16)) as RGBTriple;
+            RhAvatar._registerColor(color);
+          } else {
+            Logger.log(`[rh-avatar] invalid color ${colorCode}`);
+          }
+
+          break;
+        case 7: // ex: "#00ffcc"
+          pattern = /^#([A-f0-9]{2})([A-f0-9]{2})([A-f0-9]{2})$/.exec(colorCode);
+          if (pattern) {
+            pattern.shift();
+            const color = pattern.map(c => parseInt(c, 16)) as RGBTriple;
+            RhAvatar._registerColor(color);
+          } else {
+            Logger.log(`[rh-avatar] invalid color ${colorCode}`);
+          }
+      }
+    });
+
+    return this.colors;
+  }
+
+  get customColors() {
+    return this.css.getVariable('rh-avatar--colors');
```

Comment:
```suggestion
    return this.css.getVariable('--_colors');
```
is this meant to be private or customizable?
if private, use the [wiki convention](https://github.com/RedHat-UX/red-hat-design-system/wiki/CSS-Styles#private-variables) as above
else, drop the `--`

```suggestion
    return this.css.getVariable('rh-avatar-colors');
```

Reply:
I think this is best to be customizable, so that sites implementing can use their own colors if needed. I've updated it now!